vault/request_forwarding.go

```diff
@@ -147,6 +151,20 @@ func (c *Core) startForwarding(ctx context.Context) error {
 				// Type assert to TLS connection and handshake to populate the
 				// connection state
 				tlsConn := conn.(*tls.Conn)
+
+				// Set a deadline for the handshake. This will cause clients
+				// that don't successfully auth to be kicked out quickly.
+				// Cluster connections should be reliable so being marginally
+				// aggressive here is fine.
+				err = tlsConn.SetDeadline(time.Now().Add(10 * time.Second))
```

Comment:
Replication also uses this connection and that potentially could be going over a less reliable network. Maybe it should be bumped a little bit?

Reply:
That's a good point. 30 seconds? If it disconnects it will reconnect again...